frontend/src/components/7.Swiping/SwipingPage.js

```diff
@@ -1,31 +1,131 @@
+
+
 /**
  *
  * @return {*}
  */
 function SwipingPage() {
+  const [ButtonPopup, setButtonPopup] = useState(false);
+  const [MapPopup, setMapPopup] = useState(false);
+  const SwipedRight= [];
+
+  // Dummy data, should be retrieved by sockets
+  const RemainingTime = '25';
+  let name = 'Lonestar';
+  let location = 'Botany';
+  let cuisine = 'European';
+  let price = '$$$';
+  let rating = '4.0';
+  let id = '36472';
+
   useEffect(() => {
     document.title = 'Yes or No?';
   }, []);
 
   const hitDummyEndpoint = () => {
     axios.get('/sessions/testCard').then((response) =>{
-      console.log(response.data);
+      console.log('Dummy data:', response.data);
     });
   };
 
+  hitDummyEndpoint();
+
+  /**
+ * @param {null} Adds the current restaurant details to array which stores
+ * all right-swiped restaurants.
+ * @return {void}
+ */
+  function clickedYes() {
+    console.log('clicked yes');
+    const restaurantId = {name, id};
+    SwipedRight.push(restaurantId);
+    console.log(SwipedRight);
+    console.log(SwipedRight.length);
+    getNewCard();
+  }
+
+  /**
+ * @param {null} Retrieves new restaurant details
+ * @return {void}
+ */
+  function getNewCard() {
+    // Dummy data, should be retrieved by sockets
+    name = 'Burger King';
+    location = 'Manukau';
+    cuisine = 'Cheap food';
+    price = '$';
+    rating = '3.0';
+    id = '287376923';
+  }
+  /**
+ * @param {null} Retrieves new restaurant details
+ * @return {void}
+ */
+  function cardClicked() {
+    console.log('Card clicked');
+  }
+
   return (
     <>
-      <p>Swiping</p>
+      <h1 className='Title'> yumble</h1>
+      <h1 className='TimeCounter'> Remaining time: {RemainingTime}s</h1>
+      <div className='MakeCentre'>
+        <button
+          className='YesOrNoButton'
+          id='YesButton'
+          onClick = {clickedYes}
+        >
+            Keen!
+        </button>
+        <button
+          className='YesOrNoButton'
+          id='NoButton'
+        >
+            Nope!
+        </button>
+        <div className="SwipeCard" onClick = {cardClicked}>
+          <div className="CardImage"></div>
+          <p className='PlaceDetails' id = 'name'> {name} </p>
+          <p className='PlaceDetails' id= 'location'> {location} </p>
+          <p className='PlaceDetails' id = 'price'> {price} </p>
+          <p className='PlaceDetails' id = 'cuisine'> {cuisine} </p>
+          <p className='PlaceDetails' id = 'rating'> {rating} Rating</p>
+        </div>
+        <button
+          onClick={() => setMapPopup(true)}
+          className='BigBtn'
+          id='GoogleMaps_btn'
+        >
+          View on Google Maps
+          <Icon />
+        </button>
+        <MapModal trigger={MapPopup} setTrigger={setMapPopup} mapSrc={mapSrc}/>
+      </div>
```

Comment:
I feel like a lot of this stuff should be pulled out into a card component (so that it can be reused in the results page)